smartsim/_core/entrypoints/dragon.py

```diff
@@ -97,12 +97,12 @@ def run(dragon_head_address: str) -> None:
 
     while not SHUTDOWN_INITIATED:
         print(f"Listening to {dragon_head_address}")
-        req = dragon_head_socket.recv_json()
+        req = dragon_head_socket.recv_string()
         print(f"Received request: {req}")
-        drg_req = request_serializer.deserialize_from_json(str(req))
+        drg_req = request_registry.from_string(req)
         resp = dragon_backend.process_request(drg_req)
         print(f"Sending response {resp}", flush=True)
-        dragon_head_socket.send_json(response_serializer.serialize_to_json(resp))
+        dragon_head_socket.send_string(response_registry.to_string(resp))
```

Comment:
I wonder if we can just put a thin abstraction over the socket... I think we might avoid forcing callers to do their own conversions?

```py
def send_message(message: Message) -> None:
  wire_format = registry.to_string(Message)
  dragon_head_socket.send_string(wire_format)
```

Reply:
Makes sense to me!! I didn't love the idea of having a "mega-registry" for two main reasons: (a) it would remove a lot of the type narrowing benefits of having specific registries for specific purposes, and (b) I couldn't come up with a proper way to handle key conflicts on both ends of the wire seamlessly. That said, I see no reason our abstraction cannot simply use two registries: one for incoming and one for out going messages!

Added the `SocketSchemaTranslator` class to handle conversions of schemas to and from strings for sockets, anded some helper fns for dragon specific use of the `SocketSchemaTranslator` class, removed the `_Pipline` usage as it became obsolete, and added tests!